util/system/path_unix.go

```diff
@@ -3,8 +3,16 @@
 
 package system
 
+import (
+	"path/filepath"
+)
+
 // CheckSystemDriveAndRemoveDriveLetter verifies that a path, if it includes a drive letter,
 // is the system drive. This is a no-op on Linux.
 func CheckSystemDriveAndRemoveDriveLetter(path string) (string, error) {
```

Comment:
I noticed this function, and it looks like code was forker from moby/moby, e.g. this one came from https://github.com/moby/moby/blob/487fa4a469b4fc77e1706f14acbd97c06383e754/pkg/system/path.go#L33

Should we look at unifying these packages to prevent maintaining two distinct implementations (and at risk of diverging?)

Reply:
Sounds good to me! Any suggestions in regards to which package this would best be placed in?

Reply:
I'd have to give it a look; my eye just dropped on this one, and the name of this function looked _very_ familiar :joy:
I also may be ruffling a hornets nest here :see_no_evil:

Some quick background (might look more in-depth later);

- I know code has been forked / copied from docker to BuildKit, some of that related to there being a circular dependency between docker and buildkit (which can be "fun"), and packages in the docker repository not always being "easy" to consume (there's various packages that had dependencies on too many other things, which could cause dependency hell).
- While understandable; it's also slightly risky; implementations may start to diverge, and (due to the circular dependency) in some cases leads to _two_ implementations _both_ being used in the same codebase. If both implementations have the same (similar) requirements, we should definitely try to avoid that.
- I've been untangling some of that in the docker repository (still some work in progress), so that packages that have _known_ consumers had no (or very little) dependencies.
- In some cases, we decided to move those packages to be their own repository and module (e.g., https://github.com/moby/sys, which provides some modules, or https://github.com/moby/term)
- A separate repo (and/or module) is definitely the easiest way to consume them, but not all packages may make sense to be a "project on their own"); so that would have to be looked at.
- If it  _doesn't_ make sense as an independent module, we can still look what the best home is for it (which could be either inside the BuildKit repository, or in the Docker repository).
- We're in the process of migrating docker to be a proper go module (which will _solve_ some things, but may _complicate_ some others, e.g. having a "go.mod" can force some things and be slightly less flexible).

Reply:
What do you think about adding a new module in the https://github.com/moby/sys called something like ```path```, or ```util```. It seems that both moby and buildkit have the ```system``` package copied over. There are probably a lot more of these small packages/functions re-implemented in various places.

Should we do this as part of this PR or should we open an issue and move it as part of a different change? What do you think would be the better way?

Reply:
For the time being, I think this can live in `buildkitd`. I will be happy to move this somewhere else when we decide where the best place for it would be. I am still new to both the `buildkitd` and `moby` code base, so I am not sure where that place would be :smile: .